### PR TITLE
[luci-interpreter]Negative Tests for DepthToSpace kernel

### PR DESCRIPTION
**Pull request by struss**
for #1623 

This commit add Negative case for `DepthToSpace` kernel and update kernel implementaion.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/DepthToSpace.cpp [hunk 1]
```diff
@@ -52,9 +43,13 @@ void DepthToSpace::configure()
   int32_t output_width = input_width * block_size;
   int32_t output_channels = input_channels / block_size / block_size;
 
-  assert(input_height == output_height / block_size);
-  assert(input_width == output_width / block_size);
-  assert(input_channels == output_channels * block_size * block_size);
+  LUCI_INTERPRETER_CHECK(input_height ==
+                         output_height /
+                             block_size); // This will not make Exception, but TFlite have also.
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.cpp:48]: This code may be used to check overflow. It would be ok to remove the comments.

Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.cpp:48]: I got it. I will remove that comment!

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp [hunk 2]
```diff
@@ -55,6 +55,57 @@ TYPED_TEST(DepthToSpaceTest, SimpleCase)
   EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray(output_shape));
 }
 
+TEST(DepthToSpaceTest, Invalid_Input_Data_NEG)
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:58]:
```suggestion
TEST(DepthToSpaceTest, Invalid_Input_Shape_NEG)
```

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp [hunk 3]
```diff
@@ -55,6 +55,57 @@ TYPED_TEST(DepthToSpaceTest, SimpleCase)
...
+  DepthToSpace kernel = DepthToSpace(&input_tensor, &output_tensor, params);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST(DepthToSpaceTest, In_Out_Type_NEG)
+{
+  std::vector<float> input_data{1, 2, 3, 4, 5, 6, 7, 8};
+  Shape input_shape{1, 1, 2, 4};
+  std::vector<uint8_t> output_data{1, 2, 5, 6, 3, 4, 7, 8};
+  std::vector<int32_t> output_shape{1, 2, 4, 1};
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:80]: It seems that these are not used

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp [hunk 4]
```diff
@@ -55,6 +55,57 @@ TYPED_TEST(DepthToSpaceTest, SimpleCase)
...
+  DepthToSpace kernel = DepthToSpace(&input_tensor, &output_tensor, params);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST(DepthToSpaceTest, Block_Size_NEG)
+{
+  std::vector<float> input_data{1, 2, 3, 4, 5, 6, 7, 8};
+  Shape input_shape{1, 1, 2, 4};
+  std::vector<float> output_data{1, 2, 5, 6, 3, 4, 7, 8};
+  std::vector<int32_t> output_shape{1, 2, 4, 1};
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:97]: It seems that these are not used

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp [hunk 5]
```diff
@@ -55,6 +55,57 @@ TYPED_TEST(DepthToSpaceTest, SimpleCase)
...
+  Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+
+  DepthToSpaceParams params{};
+  params.block_size = 2;
+
+  DepthToSpace kernel = DepthToSpace(&input_tensor, &output_tensor, params);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST(DepthToSpaceTest, In_Out_Type_NEG)
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:75]:
```suggestion
TEST(DepthToSpaceTest, In_Out_Type_Mismatch_NEG)
```

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:58]: I will fix it!

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp (hunk 3 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:80]: I will remove all that are not used!

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp (hunk 4 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:97]: same as above. Thanks!

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp (hunk 5 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:75]: I will fix it! Thanks!

File: compiler/luci-interpreter/src/kernels/DepthToSpace.cpp [hunk 6]
```diff
@@ -30,20 +30,11 @@ DepthToSpace::DepthToSpace(const Tensor *input, Tensor *output, const DepthToSpa
...
-    throw std::runtime_error("Invalid output type");
-  }
-  if (input()->element_type() != output()->element_type())
-  {
-    throw std::runtime_error("Type mismatch on input and output.");
-  }
+  LUCI_INTERPRETER_CHECK(input()->shape().num_dims() == 4);
+  LUCI_INTERPRETER_CHECK(output()->element_type() == DataType::FLOAT32 ||
+                         output()->element_type() ==
+                             DataType::U8) // only execute will float32, uint8.
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.cpp:36]: The comment seems unnecessary. It is obvious from the context.

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp [hunk 7]
```diff
@@ -55,6 +55,51 @@ TYPED_TEST(DepthToSpaceTest, SimpleCase)
...
+  Tensor output_tensor = makeOutputTensor(DataType::U8);
+
+  DepthToSpaceParams params{};
+  params.block_size = 2;
+
+  DepthToSpace kernel = DepthToSpace(&input_tensor, &output_tensor, params);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST(DepthToSpaceTest, Block_Size_NEG)
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:88]:
```suggestion
TEST(DepthToSpaceTest, InvalidBlockSize_NEG)
```
BTW what's the point of adding underscores between words?)

File: compiler/luci-interpreter/src/kernels/DepthToSpace.cpp (hunk 6 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.cpp:36]: Oh I will remove this.

File: compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp (hunk 7 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/DepthToSpace.test.cpp:88]: Sorry, That's my mistake. I will fix it.

**Review by jinevening — APPROVED** [review]

**Review by s-barannikov — APPROVED** [review]

**Pull request merged**